qiskit/algorithms/evolvers/real/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,178 @@
+
+from typing import Union, Dict, Optional
+
+from qiskit.algorithms import EvolutionProblem, EvolutionResult, RealEvolver, eval_observables
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    OperatorBase,
+    SummedOp,
+    PauliOp,
+    CircuitOp,
+    ExpectationBase,
+    CircuitSampler,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.providers import Backend, BaseBackend
+from qiskit.synthesis import ProductFormula, LieTrotter
+from qiskit.utils import QuantumInstance, algorithm_globals
+from .trotter_ops_validator import is_op_bound
+
+
+class TrotterQrte(RealEvolver):
+    """ Class for performing Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Y, Zero
+            from qiskit.algorithms import EvolutionProblem, EvolutionResult
+            from qiskit.algorithms.evolvers.real.implementations.\
+                trotterization.trotter_qrte import TrotterQrte
+
+            operator = X + Z
+            initial_state = Zero
+            time = 1
+            evolution_problem = EvolutionProblem(operator, 1, initial_state)
+            # LieTrotter with 1 rep
+            trotter_qrte = TrotterQrte(evolution_problem)
+            evolved_state = trotter_qrte.evolve().evolved_state
+    """
+
+    def __init__(
+        self,
+        quantum_instance: Union[QuantumInstance, BaseBackend, Backend],
+        product_formula: ProductFormula = LieTrotter(),
+        expectation: Optional[ExpectationBase] = None,
+    ) -> None:
+        """
+        Args:
+            quantum_instance: A quantum instance used for calculations.
+            product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+                first order product formula with a single repetition.
+            expectation: An instance of ExpectationBase which defines a method for calculating
+                expectation values of EvolutionProblem.aux_operators.
+        """
+        self.product_formula = product_formula
+        self._quantum_instance = quantum_instance
+        self._expectation = expectation
+
+        self._circuit_sampler = CircuitSampler(quantum_instance)
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        return True
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        """
+        Evolves a quantum state for a given time using the Trotterization method
+        based on a product formula provided.
+        Time-dependent Hamiltonians are not yet supported.
+
+        Args:
+            evolution_problem: Instance defining evolution problem. For the included Hamiltonian,
+                only SummedOp, PauliOp are supported by TrotterQrte.
+
+        Returns:
+            Evolution result that includes an evolved state.
+
+        Raises:
```

Comment:
I think this should include the ValueError for the Initial State being required. Maybe this should be noted clearly in the docstring for the evolution problem that it must include an initial state.